### DISCUSSION TITLE
Autocomplete API

### DISCUSSION THREAD

**Discussion by its-monotype** [formaat-design/reshaped#312] [Q&A] [answered]
Is it possible to control the open/closed state or get events like onOpen? Can I render an autocomplete dropdown without absolute positioning, or use View with Menu.Item s? Additionally, can we limit the maximum height of the dropdown?

**Comment by its-monotype** [comment]
@blvdmitry please help 🙏 or maybe it's better to use cmdk or some custom solution for not absolutely positioned items more advanced use case? But still controlling an open state might be a great feature

**Reply by blvdmitry** [reply, marked as answer]
Hey, sorry, completely missed this one.

- I can definitely add ref access to the open/close methods
- Limiting maxHeight is currently possible by wrapping the items with an elements with a maxHeight and overflow auto but I can look into adding a way to achieve the same with props
- Positioning is a bit more complicated. Should be also possible to implement but in case you're doing this for a command palette, I would check first what is better – building a completely customizable solution or just building a cmdk-like separate component.

Let's move this discussion to an issue that I can pick up?

**Reply by its-monotype** [reply]
Sure, I'll add more context with use cases then
